src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -169,7 +169,7 @@ public async Task StartAsync(CancellationToken cancellationToken)
         private void CheckFileSystem()
         {
             // Shutdown if RunFromZipFailed
-            if (_environment.IsZipDeployment(validate: false))
+            if (_applicationHostOptions.CurrentValue.AreZipDeploymentAppSettingsValid)
```

Comment:
I think checking if the appsettings are not empty can be done in _environment. We can then remove AreZipDeploymentAppSettingsValid from ScriptApplicationHostOptions